snapcraft/internal/lifecycle.py

```diff
@@ -368,6 +368,9 @@ def _snap_data_from_dir(directory):
 
 
 def snap(project_options, directory=None, output=None):
+    # Check for our prerequesite external command early
+    repo.check_for_command('mksquashfs')
```

Comment:
I think that implementing in the generic `Repo` would help, so that

```python
>>> print(Ubuntu.get_installable('mksquashfs'))
Installable(package='squashfs-tools', tool='apt')
>>> print(Homebrew.get_installable('mksquashfs'))
Installable(package='squashfs', tool='brew')
>>> print(Ubuntu.get_installable('lxd'))
Installable(package='lxd', tool='snap')
```

Names and attributes are all just made up on the fly, we only need to support the tools snapcraft requires and it isn't as much of a deal on an installation from a package either.